rpc/namespaces/ethereum/debug/api.go

```diff
@@ -121,22 +112,22 @@ func (a *API) TraceTransaction(hash common.Hash, config *evmtypes.TraceConfig) (
 		}
 	}
 
-	tx, err := a.clientCtx.TxConfig.TxDecoder()(transaction.Tx)
+	tx, err := a.clientCtx.TxConfig.TxDecoder()(blk.Block.Txs[transaction.TxIndex])
```

Comment:
is this guaranteed to be within a bound?

Reply:
I think so, it's either come from our custom indexer, or the tm indexer, in both cases, they are inferred by iterating the txs in the block.